### ISSUE TITLE
Output recordings play too fast

### ISSUE THREAD

**Issue by alanimgur** [WindowsNT/ScreenCapture#2]
Thank you again for this great little library. It's truly something that other developers need; that the world needs. I'm grateful to have come across it.

I have however noticed that the recordings it produces play too fast, despite having the correct FPS. No matter what I do, or what I record, the resulting recordings playback faster than they should.

I've taken a recording of a timer on YouTube. You can see that it's counting down too fast, compared to the original:
- Recording: https://i.imgur.com/mmgdW9a.mp4
- YouTube timer: https://www.youtube.com/watch?v=4ASKMcdCc3g

To reproduce, simply build the example project with no changes and compare the output video with the YouTube timer. I've tried all the encoders available, as well as different FPS settings, with no luck.

**Comment by WindowsNT** [comment]
I will take a look as well, needs some time. What happens if you also record audio?

**Comment by ghost** [comment]
This issue still exists

**Comment by WindowsNT** [comment]
In my Turbo Play (www.turbo-play.com) I include this feature in importing screen captured videos. If you try it there, does it still have issues in speed?

**Comment by ghost** [comment]
> 
> 
> In my Turbo Play ([www.turbo-play.com](http://www.turbo-play.com)) I include this feature in importing screen captured videos. If you try it there, does it still have issues in speed?

### Yes.

https://user-images.githubusercontent.com/30274499/110411785-beb1e280-8050-11eb-864e-fe09c485aa24.mp4

**Comment by ahmed605** [comment]
Any news on this issue?

**Comment by WindowsNT** [comment]
Not yet, I 'm looking into it, perhaps scaling ThisDurV (or raising fps to 60) would be a temporary solution.

**Comment by Boomslangnz** [comment]
yes Ive got same issue here, would be great to get a fix!

*EDIT*

this resolved itself once I added Audio here

**Comment by WindowsNT** [comment]
Sorry for the delayed reply (vacation), gotta also work in HDR mode  capture.

**Comment by Boomslangnz** [comment]
> Sorry for the delayed reply (vacation), gotta also work in HDR mode capture.

Hi, thanks for response. How do i enable HDR mode capture? I dont recall seeing any setting for that

**Comment by WindowsNT** [comment]
If you have a HDR screen and in HDR mode, then you can capture a 10-bit sceen but then you have to write it in a 10-bit MP4 file, Windows Media Foundation doesn't yet support such mp4 files; I'm trying to use a NVidia Encoder directly.

**Comment by dong2gua** [comment]
I also encountered the same problem when calling this repo code to record the screen. The entire video feels sped up. Therefore, I suspect that it is caused by delay during sampling.
After carefully checking the source code, I found out. In the sampling loop, the current code does not calculate the actual time spent from the previous frame to the current frame. Instead, an ideal value is simply calculated based on fps. And set it to the Sampling Time of the current frame.
In response to the above problem, I wrote some code to fix it. And tested it in my app and it works fine.
